### PR TITLE
Create javascript bindings crate and implement order queries

### PR DESCRIPTION
**Pull request by findolor**
## Motivation

See issue: #966

## Solution

- Create another crate called `js_api` under crates.
- Implement two order queries
   - Order details
   - Orders list (multichain)

## Checks

By submitting this for review, I'm confirming I've done the following:
- [x] made this PR as small as possible
- [x] unit-tested any new functionality
- [x] linked any relevant issues or PRs
- [ ] included screenshots (if this involves a front-end change)

### CODE REVIEW

File: crates/js_api/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+description = "Javascript bindings for common functions accross the app."
+version.workspace = true
+edition.workspace = true
+license.workspace = true
+homepage.workspace = true
+
+[lib]
+crate-type = ["cdylib"]
+
+[target.'cfg(target_family = "wasm")'.dependencies]
```
Comment (rouzwelt) [inline crates/js_api/Cargo.toml:12]: we dont need cfg at all in this crate, everything will be without cfg, so normal `dependencies`

File: crates/js_api/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+[package]
+name = "rain_orderbook_js_api"
+description = "Javascript bindings for common functions accross the app."
+version.workspace = true
+edition.workspace = true
+license.workspace = true
+homepage.workspace = true
```
Comment (rouzwelt) [inline crates/js_api/Cargo.toml:7]: add a `publish = false` here so it doesnt get part of published rust crates

File: crates/subgraph/src/multi_orderbook_client.rs [hunk 3]
```diff
@@ -9,11 +10,29 @@ use crate::{
...
+    js_value_vector_from_abi, js_value_vector_into_abi, FromWasmAbi, IntoWasmAbi,
+    LongRefFromWasmAbi, RefFromWasmAbi, TryFromJsValue, VectorFromWasmAbi, VectorIntoWasmAbi,
+};
+#[cfg(target_family = "wasm")]
+use wasm_bindgen::describe::{inform, WasmDescribe, WasmDescribeVector, VECTOR};
+#[cfg(target_family = "wasm")]
+use wasm_bindgen::{JsValue, UnwrapThrowExt};
+
+#[derive(Debug, Clone, Serialize, Deserialize, Tsify)]
+#[tsify(into_wasm_abi, from_wasm_abi)]
```
Comment (rouzwelt) [inline crates/subgraph/src/multi_orderbook_client.rs:28]: do we really need these impls for `MultiSubgraphArgs`, I dont see it being used as a binding anywhere on the js-api/src/subgrapg/order.rs

File: crates/subgraph/src/pagination.rs [hunk 4]
```diff
@@ -2,8 +2,12 @@ use crate::{cynic_client::CynicClientError, utils::slice_list};
 use serde::{Deserialize, Serialize};
 use std::num::TryFromIntError;
 use thiserror::Error;
+use tsify::Tsify;
+use typeshare::typeshare;
 
-#[derive(Clone, Serialize, Deserialize, Debug)]
+#[derive(Clone, Serialize, Deserialize, Debug, Tsify)]
+#[typeshare]
+#[tsify(into_wasm_abi, from_wasm_abi)]
```
Comment (rouzwelt) [inline crates/subgraph/src/pagination.rs:10]: missing `serde rename to camelCase`

**Review by rouzwelt — CHANGES_REQUESTED** [review]

File: crates/subgraph/src/multi_orderbook_client.rs (hunk 3 above)
Comment (findolor) [reply crates/subgraph/src/multi_orderbook_client.rs:28]: This one is used as a parameter in the `getOrders` function

File: crates/js_api/src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,2 @@
+#[cfg(target_family = "wasm")]
```
Comment (rouzwelt) [inline crates/js_api/src/lib.rs:1]: not needed

**Review by rouzwelt — CHANGES_REQUESTED** [review]

**Review by rouzwelt — APPROVED** [review]

**Review by hardyjosh — APPROVED** [review]

**Pull request merged**